core/dbt/contracts/graph/metrics.py

```diff
@@ -0,0 +1,30 @@
+
+class MetricReference(object):
+    def __init__(self, metric_name, package_name=None):
+        self.metric_name = metric_name
+        self.package_name = package_name
+
+
+class ResolvedMetricReference(MetricReference):
+    """
+    Simple proxy over a ParsedMetric which delegates property
+    lookups to the underlying node. Also adds helper functions
+    for working with metrics (ie. __str__ and templating functions)
+    """
+    def __init__(self, node):
+        super().__init__(node.name, node.package_name)
+        self.node = node
+
+    def __getattr__(self, key):
+        return getattr(self.node, key)
+
+    @property
+    def is_derived(self):
+        return self.node.is_derived
+
+    def qualify(self, namespace):
+        return f'{namespace}.{self.metric_name}'
+
+    def __str__(self):
+        return self.node.name
```

Comment:
When would the ResolvedMetricReference get stringified?

Reply:
I think this would happen when you interpolate the results of the `metric()` function, so:

```
select
  {{ metric('revenue') }}
from {{ metric('revenue').model }}
```

would evaluate to
```
select
  revenue
from analytics.fct_revenue
```

So - returning the name of the metric is kind of useless, but I am unsure if we want to return anything more sophisticated than that? I guess we could return `self.node.sql` instead (which feels reasonable), but this doesn't extend well to the `ratio` metric type which does not have a `.sql` value